File: gang/week16/문자열 나누기.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        char f = s.charAt(0);   // 첫번째 문자 지정
        int x = 1;  // x는 처음부터 정해져 있으니 1로 시작
        int rX = 0;
        for(int i=1; i < s.length(); i++) {
            if(s.charAt(i) == f) {  // 문자가 같으면 x 증가
                x++;
            }
            else {  // 다르면 rX증가
                rX++;
            }
            if(x == rX) {   // 두개가 같아지면
                answer++;   // answer 증가
                if (i + 1 < s.length()) {   // 범위를 벗어나지 않으면
                    f = s.charAt(i + 1);    // 처음 문자 다시 지정
                }
                x = 0;  // 이 횟수도 초기화
                rX = 0;
            }
        }
        if(x != rX) {   // 반복문을 다 돌렸는데 같지 않으면 나머지도 문자열이 되기 때문에 answer 증가
            answer++;
        }
        return answer;
    }
}

File: jeonghyeon/16week/문자열 나누기.js
function solution(s) {
    let answer = 0;
    let first = 0; // 현재 문자 개수
    let next = 0; // 다른 문자 개수
    let current = ""; // 현재 문자열
    for(let i = 0; i < s.length; i++) {
        if(current === "") { // 공백이면
            current = s[i]; // 현재 문자를 저장
            first++;
        } else if(s[i] === current) { // 현재 본 글자와 저장된 문자열이 같다면 
            first++;
        } else next++; // 다르면 다른 문자 개수 증가
        
        if(first === next) { // 현재 문자와 다른 문자 개수가 같아지면 
            answer++; 
            current = "";
            first = 0;
            next = 0; // 초기화
        }
    }
    if(current !== "") { // 공백으로 s가 끝나지 않았을 때
        answer++;
    }
    return answer;
}

// 처음 나온 문자랑 다른 문자들은 다른 문자의 개수를 증가 aaabbacc 는 aaaa 4개와 그 외 4개로 이루어짐